### ISSUE TITLE
Couldn‘t find device when I run POCL on RISC-V QEMU

### ISSUE THREAD

**Issue by dinyy** [pocl/pocl#1372]
It shows CL_DEVICE_NOT_FOUND when I run ./example0 ,like this pic
![pic_ex](https://github.com/pocl/pocl/assets/107465691/d1f5907d-416a-4e16-9bda-89c6be9f7283)
![pic_2](https://github.com/pocl/pocl/assets/107465691/75f377cf-9ae5-48ad-96b1-37fc9b488b9a)

**Comment by dinyy** [comment]
When I build PoCL，occur a problem is that：host cpu is unknown（LLVM couldn't identify）
![pic_3](https://github.com/pocl/pocl/assets/107465691/c03f5fdd-f9f4-496d-8379-e744cf89b05a)
but I could get my cpu info:
![pic_4](https://github.com/pocl/pocl/assets/107465691/bd3de3bc-9856-461b-b782-c1728d961f3f)
mybe I miss some message？

**Comment by jansol** [comment]
Did you set `POCL_BUILDING=1` in your environment? It looks like you are trying to run from the build directory and that will fail if that variable is not set.

**Comment by pjaaskel** [comment]
Also pls check https://github.com/pocl/pocl/blob/main/doc/sphinx/source/install.rst#risc-v-support, the notes related to the CPU detection. Please let us know if you manage to make it run!

**Comment by dinyy** [comment]
> Did you set `POCL_BUILDING=1` in your environment? It looks like you are trying to run from the build directory and that will fail if that variable is not set.

I set it. Let me try again on ubuntu23.

**Comment by franz** [comment]
I suspect LLVM 12 is too old for RISC-V. But it looks like PoCL is loaded which is good. You can try to run with `POCL_DEBUG=all ./example0` and it should print some more information on why it can't use the CPU device.

**Comment by dinyy** [comment]
> I suspect LLVM 12 is too old for RISC-V. But it looks like PoCL is loaded which is good. You can try to run with `POCL_DEBUG=all ./example0` and it should print some more information on why it can't use the CPU device.

I haven't encountered this issue when using Ubuntu 23 on QEMU with LLVM-15.But I still couldn't run it successfully. I  meet the problem
 "can't link double-float modules with soft-float modules"
,although I use -DHOST_CPU_TARGET_ABI=lp64  for generic rv64. Is there a problem with my flag?

**Comment by franz** [comment]
@dinyy  did you run with clean pocl cache ? you can check if the `-target-cpu` and `-target-abi` flags are passed correctly, by running any opencl example or test with `POCL_DEBUG=llvm` (pocl cache must be clean to trigger recompilation). It should print something like `LLVM | all build options: ..... -triple=X-Y-Z -target-cpu XYZ -target-abi XYZ`

**Comment by dinyy** [comment]
> @dinyy did you run with clean pocl cache ? you can check if the `-target-cpu` and `-target-abi` flags are passed correctly, by running any opencl example or test with `POCL_DEBUG=llvm` (pocl cache must be clean to trigger recompilation). It should print something like `LLVM | all build options: ..... -triple=X-Y-Z -target-cpu XYZ -target-abi XYZ`

 thanks，I only find `-triple=riscv64-unknown-linux-gnu -target-cpu generic-rv64` and no -target-abi ,maybe I should check why the ` -DHOST_CPU_TARGET_ABI=lp64d `flag not take effect.

**Comment by dinyy** [comment]
Indeedly thank you so much，I have solved my problom refferring this  [issue](https://github.com/pocl/pocl/issues/1088).
